File: Programmers/Python/숫자 문자열과 영단어.py
def solution(s):
    answer = 0
    numList = []
    strList = list(s)
    word = {
        "zero": "0",
        "one": "1",
        "two": "2",
        "three": "3",
        "four": "4",
        "five": "5",
        "six": "6",
        "seven": "7",
        "eight": "8",
        "nine": "9"
    }

    temp = ""
    for string in strList:
        if string.isdigit():
            numList.append(string)
        else:
            temp += string
            if temp in word:
                numList.append(word[temp])
                temp = ""

    answer = int(''.join(numList))

    return answer


"""
hash map (= dictionary) 에 미리 저장해둔 단어를 매칭
"""

File: Programmers/JavaScript/숫자 문자열과 영단어.js
function solution(s) {
    let answer = s;
    let word = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]

    for(let i=0; i<word.length; i++){
        let arr = answer.split(word[i]);
        answer = arr.join(i)
    }

    answer = Number(answer)
    return answer;
}

/*
단어로 split 한 후, 해당 단어를 나타내는 숫자로 다시 이어 붙혀주는 방식을 사용.
 */